test/network/BancorNetwork.ts

```diff
@@ -2394,13 +2483,19 @@ describe('BancorNetwork', () => {
                 pendingNetworkFeeAmount = pendingNetworkFeeAmount.add(hop1.networkFeeAmount.add(hop2.networkFeeAmount));
             }
 
-            const res = await tradeFunc(amount, {
-                limit,
-                beneficiary: beneficiaryAddress,
-                deadline
-            });
+            const { retVal, res } = await tradeFunc(
+                amount,
+                {
+                    limit,
+                    beneficiary: beneficiaryAddress,
+                    deadline
+                },
+                true
+            );
+
+            expect(retVal).to.equal(hop2.amount);
 
-            const transactionCost = await getTransactionCost(res);
+            const transactionCost = await getTransactionCost(await res);
```

Comment:
Let's `await` in `tradeFunc` and then its users won't need to remember to do it

Reply:
You can't do that, because it sometimes reverts (depending on the test).
That's probably why the author of these tests has returned it unresolved to begin with.

Reply:
Solution revised